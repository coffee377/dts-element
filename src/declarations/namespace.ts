import { Declaration, IDeclarationOptionalParameters } from '../declaration';
import { indent } from '../helpers/indent';

export interface INamespaceOptionalParameters {
  declarations: Declaration<any, any>[];
}

export class Namespace extends Declaration<{}, INamespaceOptionalParameters> {

  public emit(): string {
    const jsdoc = this.emit_jsdoc();
    const content = this.parameters.declarations
      .map((declaration: Declaration<any, any>) => declaration.emit())
      .join('\n');
    return `${jsdoc}declare namespace {\n${indent(content)}}`;
  }

  protected get_default_parameters(): IDeclarationOptionalParameters & INamespaceOptionalParameters {
    return Object.assign({}, super._get_default_parameters(), {
      declarations: [],
    });
  }

}
