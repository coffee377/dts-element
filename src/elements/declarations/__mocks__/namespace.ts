import {Declaration} from '../../declaration';

export class NamespaceDeclaration extends Declaration<any, any> {

  public get default_parameters(): any {
    return super.default_declaration_parameters;
  }

  public _emit_raw(): string {
    return `[NamespaceDeclaration ${this.parameters.name}]`;
  }

}
