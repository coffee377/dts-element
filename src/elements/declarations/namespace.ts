import {Container} from '../../collections';
import {emit_declaration_name} from '../../helpers/emit-declaration-name';
import {emit_declare} from '../../helpers/emit-declare';
import {emit_elements} from '../../helpers/emit-elements';
import {indent_every_line} from '../../helpers/indent-every-line';
import {Declaration, IDeclarationOptionalParameters} from '../declaration';

// tslint:disable-next-line no-empty-interface
export interface INamespaceDeclarationRequiredParameters {}

export interface INamespaceDeclarationOptionalParameters {
  children: Declaration[];
}

export class NamespaceDeclaration
    extends Declaration<INamespaceDeclarationRequiredParameters, INamespaceDeclarationOptionalParameters> {

  public get default_parameters(): IDeclarationOptionalParameters & INamespaceDeclarationOptionalParameters {
    return Object.assign({}, super.default_declaration_parameters, {
      children: [],
    });
  }

  public _emit_raw(container: Container): string {
    const name = emit_declaration_name(this.parameters.name, container);
    const content = emit_elements(this.parameters.children, this);
    const declare = emit_declare(container);
    return `${declare}namespace ${name} {\n${indent_every_line(content)}\n}`;
  }

}
