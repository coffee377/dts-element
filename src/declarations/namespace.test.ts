import { Namespace } from './namespace';

describe('#emit()', () => {
  it('should return formatted namespace', () => {
    const a_namespace = new Namespace({ name: 'Test', jsdoc: 'Line 1\nLine 2' });
    expect(a_namespace.emit()).toMatchSnapshot();
  });
});
