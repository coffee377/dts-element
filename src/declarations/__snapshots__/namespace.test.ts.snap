// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#emit() should return formatted namespace 1`] = `
"/**
 * Line 1
 * Line 2
 */
declare namespace Test {
}"
`;
